### PR TITLE
[ci-daily] Fix pipeline tests

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Should fix the `automatic_speech_recognition_pipeline` tests. 
Also using `streaming` dataset to speed up tests. Think it is a good idea if we are only using 1 data.

### CODE REVIEW

File: src/transformers/pipelines/automatic_speech_recognition.py [hunk 1]
```diff
@@ -635,9 +640,9 @@ def postprocess(self, model_outputs, decoder_kwargs: Optional[Dict] = None, retu
                 # Simply cast from pyctcdecode format to wav2vec2 format to leverage
                 # pre-existing code later
                 chunk_offset = beams[0][2]
-                word_offsets = []
+                offsets = []
                 for word, (start_offset, end_offset) in chunk_offset:
-                    word_offsets.append({"word": word, "start_offset": start_offset, "end_offset": end_offset})
+                    offsets.append({"word": word, "start_offset": start_offset, "end_offset": end_offset})
```
Comment (ArthurZucker) [inline src/transformers/pipelines/automatic_speech_recognition.py:645]: normalized the name of this variable

File: src/transformers/pipelines/automatic_speech_recognition.py [hunk 2]
```diff
@@ -520,12 +526,11 @@ def preprocess(self, inputs, chunk_length_s=0, stride_length_s=None, ignore_warn
                 processed["stride"] = stride
             yield {"is_last": True, **processed, **extra}
 
-    def _forward(self, model_inputs, generate_kwargs=None):
+    def _forward(self, model_inputs, return_timestamps=False, generate_kwargs=None):
```
Comment (ArthurZucker) [inline src/transformers/pipelines/automatic_speech_recognition.py:529]: Adding this argument prevent the `.pop` from removing it for other processes.

File: src/transformers/pipelines/automatic_speech_recognition.py [hunk 3]
```diff
@@ -68,9 +68,15 @@ def chunk_iter(inputs, feature_extractor, chunk_len, stride_left, stride_right,
...
+
         chunk_len = chunk.shape[0]
         stride = (chunk_len, _stride_left, _stride_right)
-        if ratio != 1:
+        if "input_features" in processed:
+            processed_len = processed["input_features"].shape[-1]
+        elif "input_values" in processed:
+            processed_len = processed["input_values"].shape[-1]
+        if processed_len != chunk.shape[-1] and rescale:
+            ratio = processed_len / chunk_len
```
Comment (ArthurZucker) [inline src/transformers/pipelines/automatic_speech_recognition.py:79]: This was missing! Fixes the LM tests

**Review by sgugger — APPROVED** [review]
LGTM if @Narsil agrees :-)

**Review by Narsil — APPROVED** [review]

**Pull request merged**